packages/vscode-extension/src/webview/components/UrlSelect.tsx

```diff
@@ -32,16 +96,37 @@ function UrlSelect({ onValueChange, items, value, disabled }: UrlSelectProps) {
           className="url-select-content"
           position="popper"
           style={{ width: urlWidth }}>
+          <Select.ScrollUpButton className="url-select-scroll">
+            <span className="codicon codicon-chevron-up" />
+          </Select.ScrollUpButton>
           <Select.Viewport className="url-select-viewport">
-            {items.map(
-              (item) =>
-                item.name && (
-                  <SelectItem value={item.id} key={item.id}>
-                    {item.name}
-                  </SelectItem>
-                )
-            )}
+            <Select.Group>
+              <Select.Label className="url-select-label">Recently used:</Select.Label>
+              {recentItems.map(
+                (item) =>
+                  item.name && (
+                    <SelectItem value={`recent#${item.id}`} key={item.id} urlWidth={urlWidth}>
+                      {item.name.length > maxCharsInSingleLine ? splitLines(item.name) : item.name}
```

Comment:
Why we can't achieve this with CSS?

Reply:
I've managed to remove the computation and passing of urlWidth and use css to determine components' width.